src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -430,8 +437,8 @@ partial class BoundLiteral : ILiteralExpression
 
     partial class BoundObjectCreationExpression : IObjectCreationExpression
     {
-        private static readonly ConditionalWeakTable<BoundObjectCreationExpression, object> s_memberInitializersMappings =
-            new ConditionalWeakTable<BoundObjectCreationExpression, object>();
+        private static readonly ConditionalWeakTable<BoundObjectCreationExpression, StrongBox<ImmutableArray<IMemberInitializer>>> s_memberInitializersMappings =
+            new ConditionalWeakTable<BoundObjectCreationExpression, StrongBox<ImmutableArray<IMemberInitializer>>>();
 
```

Comment:
I guess not. What is the downside of StrongBox? From the memory allocation point of view, I don't see much difference though.